core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -252,10 +252,8 @@ private[zio] final class FiberContext[E, A](
               // Cannot capture `curZio` since it will be boxed into `ObjectRef`,
               // which destroys performance. So put `curZio` into a temp val:
               val tmpIo = curZio
-
-              curZio = ZIO.yieldNow *> tmpIo
-
-              opcount = 0
+              evaluateLater(tmpIo)
```

Comment:
This solution does not require `tmpIo`, so you can simplify to:

```scala
evaluateLater(tmpIo)
curZio = null
```

Reply:
done, thank you.